Bluetooth: Mesh: Shell improvements

PavelVPV opened the pull request:
This PR adds the following improvements to mesh shell:
- Moves shell Kconfig options to `mesh/shell/Kconfig` file
- Adds Kconfig options for disabling mesh shell commands for specific models and features
- Adds a macro that wraps creation of shell commands for model instance selection
- Removes `_cli` postfix for files under `mesh/shell/`

PavelVPV committed:
Bluetooth: Mesh: Move mesh shell Kconfig options to a separate file

Move mesh shell related options to a new file to avoid cluttering of
the main Bluetooth Mesh Kconfig file when extending it with new options.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV committed:
Bluetooth: Mesh: Add options to disable health and cfg clients shell

Add Kconfig options to be able to disable Health and Config Client
models shell commands.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV committed:
Bluetooth: Mesh: Make mesh features pluggable in shell

Not all shell commands may be needed for an end application even if a
certain feature is enabled. In that case they will just occupy flash
space without being used. This commit adds a separate Kconfig for each
feature to make possible to selectively disable some mesh shell commands.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV committed:
Bluetooth: Mesh: Add macro for instance set/get-all mesh shell cmds

This code is a boulerplate that will be needed for many models in mesh
shell module. This commit adds a special macro designed to improve
readability of the code and helps to avoid potential bugs when
copy-pasting identical code.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

PavelVPV committed:
Bluetooth: Mesh: Remove _cli postfix from models shell files

This commits renames `health_cli.c` and `cfg_cli.c` under `mesh/shell`
folder to `health.c` and `cfg.c` accordingly. This is to create a
generic pattern for the future, when not only clients, but server models
will also expose some API through shell commands. This means that server
models' shell commands should be located in the same file as client
commands to avoid cluttering `mesh models` subcommand.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

jhedberg commented:
Is the `_cmds` suffix really necessary for the new files? A shell module is essentially all about commands. So I’d consider just `health.c` and `cfg.c`, if you think that makes sense.

alxelax commented on subsys/bluetooth/mesh/shell/shell.c:
```diff
@@ -199,125 +199,199 @@ struct bt_mesh_health_cli bt_mesh_shell_health_cli = {
+	ctx_shell = sh;
+	shell_print(sh, "Mesh shell initialized");
+
+	(void)ctx_shell;
```
why is it required? `ctx_shell` and `default_key` are global and in use.

PavelVPV replied:
I think this was needed when mesh shell was enabled without any additional options. Not needed now, I changed linkage of these 2 to make them usable in other places.

PavelVPV committed:
Bluetooth: Mesh: Change linkage for ctx_shell and default_key in shell

Change linkage for ctx_shell and default_key in shell.c to external to
share their usage in other places.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>
